packages/2019-disaster-game/src/components/Game/TaskScreen/VoteMapScreen.js

```diff
@@ -28,39 +28,76 @@ const screenLayout = css`
 const VoteMapScreen = ({
   activeTask,
   activeTaskId,
-  completedTasks = [],
-  tasks,
+  taskLocations,
   taskVotes,
   mapTransitionDuration
 }) => {
   const [longitude, setLongitude] = useState(initialLon);
   const [latitude, setLatitude] = useState(initialLat);
+  const [data, setData] = useState(asGeoJSON(taskLocations));
 
-  const data = asGeoJSON(tasks, activeTask, completedTasks);
+  const getIcon = d => d.properties.id;
+
+  // Returns a location that hasn't been completed if possible
+  const getActiveTaskLocation = useCallback(
+    currentTask => {
+      let possibleTaskLocation = null;
+      for (let i = 0; i < taskLocations.length; i += 1) {
+        const taskAtLocation = taskLocations[i];
+        if (taskAtLocation.type === currentTask.id) {
+          if (taskAtLocation.completed) {
+            possibleTaskLocation = taskAtLocation.location;
+          } else {
+            possibleTaskLocation = taskAtLocation.location;
+            break;
+          }
+        }
+      }
+      return possibleTaskLocation;
+    },
```

Comment:
Feel free to ignore this, but I wanted to see what this would look like with `filter` and `find`.

```js
currentTask => {
  const activeLocations = taskLocations.filter(t => t.type === currentTask.id);
  if (!activeLocations.length) return null;

  const location = activeLocations.find(t => !t.completed) || activeTasks[activeTasks.length - 1]
  return location.location;
}